### PR TITLE
fix(migrations) avoid performing full updates

### PR DESCRIPTION
**Pull request by thibaultcha**
### Summary

The previous intent was to rely on a schema's default field to insert
default values for new fields. However, a `full` update means we pass a
complete row to the DAO, and that row might contain older fields. Those
fields come from the row in the DB, but **disappeared** from the schema
definition. Hence, the DAO errors on the update and complains that
fields (such as `strip_request_path`) do not exist anymore, which is
correct, according to the current API schema.

Hence, we prefer not performing full updates, and only pass a subset of
hard-coded entities' fields to `dao:update()`.

### Full changelog

* avoid performing full updates with now non-existing fields in 
`dao:update()` from migrations

### CODE REVIEW

**Comment by Tieske** [comment]
previously we relied on the schema for defaults, so we wouldn't have magic values in multiple places (defaults in but schema files as well as the migrations). But the circular reference as pointed out by @thibaultcha is a bigger evil imo.

So imo past-migrations should never be impacted by future changes. Hence we should hardcode the defaults in the migrations.

**Comment by thibaultcha** [comment]
One more reason is that such defaults can also change in the future, but those migrations should insert the current defaults.

**Comment by Tieske** [comment]
>but those migrations should insert the current defaults

imo that is overkill, say three versions;

1. without `myField`
2. with `myField` default being 2
3. with `myField` default being 3

migrating from 1 upwards should result in `myField == 2`, new entries added afterwards should get `default == 3`. Migrations `1 => 2` and `2 => 3` should be deterministic, and the results should be as static as possible. You cannot do that if older migrations start applying newer defaults.

Otherwise it would be odd imo (and very hard to maintain). What if a type changes, old default being `2`, new default being `hello`?

In short: a migration cannot be responsible for future changes it does not yet know.

**Comment by thibaultcha** [comment]
I mean the current default, **hard-coded**. You know,as it was before :)
